### DISCUSSION TITLE
Not Receiving User Picture with OpenID Connect

### DISCUSSION THREAD

**Discussion by hopewise** [zitadel/zitadel#7583] [Q&A] [answered]
Hi everyone,

I'm using OpenID Connect with Zitadel to authenticate users in my app. I've included openid email profile in the scope. However, I'm not getting the user's picture in the response. I've tried checking both the ID token and the userinfo endpoint but no luck.

Here's what I'm doing:

```
scope = 'openid email profile'
```

I successfully get other profile information, just not the picture. I also noticed when users sign in with their Gmail account, the picture is not claimed as below:
![Screenshot 2024-03-16 at 6 08 32 PM](https://github.com/zitadel/zitadel/assets/1061157/60347ece-9cc7-4335-8c26-49527184e94a)

I've included the profile scope with the required scopes in Google provider as:
![Screenshot 2024-03-16 at 6 12 22 PM](https://github.com/zitadel/zitadel/assets/1061157/7debfb68-4744-46ae-93ad-d4b0ed8f77a8)

Any idea why this might be happening or what I should check?

Thanks for your help!

**Comment by hifabienne** [comment]
I think we do not return the avatar in the profile at the moment.
@livio-a @muhlemmer is that correct?

**Reply by livio-a** [reply]
I think the problem boils down to that we store the avatars in ZITADEL and would need to retrieve them from the IdP, which is not done currently.

**Reply by hifabienne** [reply]
I think we had some community members with the same problem.
They now added some ZITADEL Actions to get the avatar from the idp to ZITADEL.
You can find the conversation here: https://discord.com/channels/927474939156643850/1146743678312923206/1157193246926196797

**Comment by hopewise** [comment]
Thank you @livio-a  @hifabienne , can you point me to how to use those actions? so those are at repo code level ? do I have to change any code? Any tutorial?

**Reply by hifabienne** [reply]
I think I just realized that it doesn't do what you expect. The actions do fill the picture claim in the tokens but not have it in ZITADEL imported. So they would not show in our user list, only for authenticated users itself

**Comment by hopewise** [comment]
That would be good also for my case, but how can I install and consume those actions ?

**Reply by hifabienne** [reply, marked as answer]
We do have a guide on how to add actions here: https://zitadel.com/docs/guides/manage/console/actions
In the examples itself you can see in the comments to which flow and trigger you have to add them: https://github.com/zitadel/actions/blob/main/examples/add_picture_claim_from_idp_metadata.js
https://github.com/zitadel/actions/blob/main/examples/set_idp_picture_metadata.js

**Comment by hopewise** [comment]
Thanks
